apps/okreads-e2e/src/integration/reading-list.spec.ts

```diff
@@ -11,4 +11,29 @@ describe('When: I use the reading list feature', () => {
       'My Reading List'
     );
   });
+
+  it('Then: I should be able to undo remove book', () => {
+    cy.get('input[type="search"]').type('angular');
+
+    cy.get('form').submit();
+
+    let isClicked = false;
+    cy.get('[data-testing="cta-add-to-reading-list"]').each((ele, index, list) => {
+      if (ele[0].innerText === 'Want to Read' && !isClicked) {
+        ele.click();
+        isClicked = true;
+      }
+    });
+    cy.wait(1000);
```

Comment:
I don't think for loop is required. You should just find  first which is not clicked yet